thunder/rdds/fileio/readers.py

```diff
@@ -626,8 +628,11 @@ def mode(self):
 def getByScheme(dataPath, lookup, default):
     """Helper function used by get*ForPath().
     """
-    parseresult = urlparse.urlparse(dataPath)
-    clazz = lookup.get(parseresult.scheme, default)
+    if hasattr(dataPath, '__iter__'):
```

Comment:
This implementation only works for a local file system, not S3, right? We can limit this PR to the local case, but would the S3 case be useful to have as well?